controller/internal/routes/base_router.go

```diff
@@ -482,185 +485,74 @@ func listWithId(ae *env.AppEnv, rc *response.RequestContext, idType response.IdT
 	rc.RequestResponder.RespondWithOk(results, meta)
 }
 
-type ListAssocF func(string, func(model.BaseModelEntity)) error
-
-func ListAssociations(ae *env.AppEnv, rc *response.RequestContext, idType response.IdType, listF ListAssocF, converter ModelToApiMapper) {
-	id, err := rc.GetIdFromRequest(idType)
-
-	if err != nil {
-		log := pfxlog.Logger()
-		logErr := fmt.Errorf("could not find id property: %v", response.IdPropertyName)
-		log.WithField("property", response.IdPropertyName).Error(logErr)
-		rc.RequestResponder.RespondWithError(err)
-		return
-	}
-
-	var modelResults []model.BaseModelEntity
-	err = listF(id, func(entity model.BaseModelEntity) {
-		modelResults = append(modelResults, entity)
-	})
+// type ListAssocF func(string, func(models.Entity)) error
+type listAssocF func(rc *response.RequestContext, id string, queryOptions *QueryOptions) (*QueryResult, error)
 
-	if err != nil {
-		if util.IsErrNotFoundErr(err) {
-			rc.RequestResponder.RespondWithNotFound()
-			return
+func ListAssociationWithHandler(ae *env.AppEnv, rc *response.RequestContext, idType response.IdType, lister models.EntityRetriever, associationLoader models.EntityRetriever, mapper ModelToApiMapper) {
+	ListAssociations(rc, idType, func(rc *response.RequestContext, id string, queryOptions *QueryOptions) (*QueryResult, error) {
+		// validate that the submitted query is only using public symbols. The query options may contain an final
+		// query which has been modified with additional filters
+		query, err := queryOptions.getFullQuery(associationLoader.GetStore())
```

Comment:
Does the error read the same between a private symbol and an unknown symbol? They should be indistinguishable otherwise it is a way to map the data model externally.

Reply:
Fixed: https://github.com/netfoundry/ziti-foundation/pull/43/commits/73ba308cb6124db4e33039ed640018d8d09ccac8